<launch>
  <arg name="output" default="screen"/>

  <node name="predicator_fake_class_node"
    type="fake_classes.py"
    pkg="predicator_fake_classification"
    output="$(arg output)">

    <rosparam param="class_info">
      - name: robot
        members:
          - wam
          - wam2
      - name: object
        members:
          - peg1
          - peg2
          - ring1
          - wam
          - wam2
          - stage
      - name: peg
        members:
          - peg1
          - peg2
      - name: ring
        members:
        - ring1

    </rosparam>

    <rosparam param="param_class_info">
      - name: component
        parent: wam
        members:
          - wam/shoulder_yaw_link
          - wam/shoulder_pitch_link
          - wam/upper_arm_link
          - wam/forearm_link
          - wam/wrist_yaw_link
          - wam/wrist_pitch_link
          - wam/wrist_palm_link
      - name: component
        parent: wam2
        members:
          - wam2/shoulder_yaw_link
          - wam2/shoulder_pitch_link
          - wam2/upper_arm_link
          - wam2/forearm_link
          - wam2/wrist_yaw_link
          - wam2/wrist_pitch_link
          - wam2/wrist_palm_link
      - name: component
        parent: peg1
        members:
          - peg1/peg_link
          - peg1/base_link
      - name: component
        parent: peg2
        members:
          - peg2/peg_link
          - peg2/base_link
      - name: component
        parent: ring1
        members:
          - ring1/ring_link
      - name: component
        parent: stage
        members:
          - stage_link
    </rosparam>


  </node>
</launch>
